src/test/java/com/adyen/PaymentTest.java

```diff
@@ -475,18 +456,36 @@ protected void assertAuthorised(PaymentResult paymentResult) {
     protected void assertRefused(PaymentResult paymentResult) {
         assertEquals(PaymentResult.ResultCodeEnum.REFUSED, paymentResult.getResultCode());
     }
+    @Test
+    public void TestByteArrayDeserialization() throws Exception {
+
+        Client client = createMockClientFromFile("mocks/authorise-success.json");
+        Payment payment = new Payment(client);
+
+        final String expectedBytesAsString = "Let's pretend this a jpg or something";
+        final byte[] expectedBytes = expectedBytesAsString.getBytes(StandardCharsets.UTF_8);
+        final ByteString expectedByteString = ByteString.of(expectedBytes);
+        final String serializedBytes = expectedByteString.base64();
+        final String serializedBytesWithQuotes = "\"" + serializedBytes + "\"";
+        Type type = new TypeToken<byte[]>() { }.getType();
+
+        // Act
+        byte[] actualDeserializedBytes = JSON.deserialize(serializedBytesWithQuotes, type);
+
+        // Assert
+        assertEquals(expectedBytesAsString, new String(actualDeserializedBytes, StandardCharsets.UTF_8));
+    }
 
     @Test
-    @Ignore
     public void TestByteArrayToJSONString() throws Exception {
         Client client = createMockClientFromFile("mocks/authorise-success.json");
         Payment payment = new Payment(client);
         PaymentRequest paymentRequest = new PaymentRequest();
         paymentRequest.mpiData(new ThreeDSecureData().cavv("AQIDBAUGBwgJCgsMDQ4PEBESExQ=".getBytes()));
         
         payment.authorise(paymentRequest);
-        
-        String expected = "\"mpiData\":{\"cavv\":\"AQIDBAUGBwgJCgsMDQ4PEBESExQ=\"}";
+        // Unicode representation of the string
+        String expected = "\"mpiData\":{\"cavv\":\"AQIDBAUGBwgJCgsMDQ4PEBESExQ\\u003d\"}";
```

Comment:
Does the API expect this format?

Reply:
This is unicode, which apparently most APIs would be able to handle just fine. @michaelpaul do you think our API's handle this? I can alternatively remove this by ignoring the html escaping in the gsonbuilder.

Reply:
Just verified with actual request in api logs, API does properly parse this string with the html escaping like that!

Reply:
Here it's our previous encoder: https://github.com/Adyen/adyen-java-api-library/blob/311eea70f6515477abe1d9a29db091ab15e2a025/src/main/java/com/adyen/serializer/ByteArrayToStringAdapter.java#L40

I think we can proceed with this new one since its very similar.